packages/flutter_tools/test/commands.shard/hermetic/screenshot_command_test.dart

```diff
@@ -26,42 +27,89 @@ void main() {
         throw Exception('dummy');
       };
 
-      await expectLater(() => createTestCommandRunner(ScreenshotCommand())
+      await expectLater(() => createTestCommandRunner(ScreenshotCommand(fs: MemoryFileSystem.test()))
         .run(<String>['screenshot', '--type=skia', '--observatory-url=http://localhost:8181']),
         throwsA(isException.having((Exception exception) => exception.toString(), 'message', contains('dummy'))),
       );
 
-      await expectLater(() => createTestCommandRunner(ScreenshotCommand())
+      await expectLater(() => createTestCommandRunner(ScreenshotCommand(fs: MemoryFileSystem.test()))
         .run(<String>['screenshot', '--type=rasterizer', '--observatory-url=http://localhost:8181']),
         throwsA(isException.having((Exception exception) => exception.toString(), 'message', contains('dummy'))),
       );
     });
 
 
     testUsingContext('rasterizer and skia screenshots require observatory uri', () async {
-      await expectLater(() => createTestCommandRunner(ScreenshotCommand())
+      await expectLater(() => createTestCommandRunner(ScreenshotCommand(fs: MemoryFileSystem.test()))
         .run(<String>['screenshot', '--type=skia']),
         throwsToolExit(message: 'Observatory URI must be specified for screenshot type skia')
       );
 
-      await expectLater(() => createTestCommandRunner(ScreenshotCommand())
+      await expectLater(() => createTestCommandRunner(ScreenshotCommand(fs: MemoryFileSystem.test()))
         .run(<String>['screenshot', '--type=rasterizer',]),
         throwsToolExit(message: 'Observatory URI must be specified for screenshot type rasterizer'),
       );
     });
 
     testUsingContext('device screenshots require device', () async {
-      await expectLater(() => createTestCommandRunner(ScreenshotCommand())
+      await expectLater(() => createTestCommandRunner(ScreenshotCommand(fs: MemoryFileSystem.test()))
         .run(<String>['screenshot']),
         throwsToolExit(message: 'Must have a connected device for screenshot type device'),
       );
     });
 
     testUsingContext('device screenshots cannot provided Observatory', () async {
-      await expectLater(() => createTestCommandRunner(ScreenshotCommand())
+      await expectLater(() => createTestCommandRunner(ScreenshotCommand(fs: MemoryFileSystem.test()))
         .run(<String>['screenshot',  '--observatory-url=http://localhost:8181']),
         throwsToolExit(message: 'Observatory URI cannot be provided for screenshot type device'),
       );
     });
   });
+
+  group('Screenshot file validation', () {
```

Comment:
Since all this command does now is check if the file exists, you only need a test case for 1) file exists and 2) file doesn't exist